services/healthcheck.go

```diff
@@ -0,0 +1,60 @@
+package services
+
+import (
+	"crypto/tls"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	"github.com/rancher/rke/hosts"
+	"github.com/sirupsen/logrus"
+)
+
+func runHealthcheck(host *hosts.Host, port int, useTLS bool, serviceName string, healthcheckDialerFactory hosts.DialerFactory) error {
+	logrus.Infof("[healthcheck] Start Healthcheck on service [%s] on host [%s]", serviceName, host.Address)
+	var logErr error
+	proto := "http"
+	for retries := 0; retries <= 5; retries++ {
+		client, err := getHealthCheckHTTPClient(host, port, healthcheckDialerFactory)
```

Comment:
Do you want to sleep between the retries? Also can we extract the code called from the for loop?

Reply:
Ok, I see that we do sleep in one of the errors. I think we should sleep in any error case. Also is there a reason the sleep is 5 seconds? sleeping for 25 seconds on per service basis seems a bit extreme to me.

Reply:
got it, i will make this change